app/controllers/tag_controller.rb

```diff
@@ -300,7 +300,7 @@ def create
                            body: "@#{current_user.username} awards a #{barnstar_info_link} to #{node.user.name} for their awesome contribution!")
 
         elsif tagname.split(':')[0] == "with"
-          user = User.find_by_username_case_insensitive(tagname.split(':')[1])
+          user = User.find_by_username_case_insensitive(current_user.username)
```

Comment:
Power-tag `with:user1` adds user with username `user1` as co-author of the note. So, we want to get their username from tag and so this change is wrong

Reply:
Ahh!!! I missed out this. Sorry for the wrong change. I'll commit with correct one soon.
Thanks @gauravano